game.py

```diff
@@ -0,0 +1,302 @@
+    if human == 'Y' or human == 'y':
+        return True
+    else:
+        return False
+
+def get_human_move(game_state):
+    valid = False
+    while not valid:
+        row = int(input("Enter row : ")) - 1
+        column = int(input("Enter column : ")) - 1
+        if  game_state.valid_move((row, column)):
+            valid = True
+        else:
+            print('Invalid move. Try again')
+            valid = False
+    return row, column
+
+def get_hint():
+    hint = input("Get hint? (Y/N) : ")
+    if hint == 'Y' or hint == 'y':
+        return True
+    else:
+        return False
+
+
+def main():
+    board = []
+    board_size = choose_board_size()
+    print('board size : ', board_size)
+    board = create_board(board_size)
+    print('board : ', board)
+    
+    player_next = 1
+    number_of_moves = 0
+    
+    global score_count
+    
+    game_state = GameState(board, player_next)
+   
+    another_move = True
+
+    while another_move:
+
+        if  human_to_play():
+            while get_hint():
+                score_count = 0
+                best_move_and_score = get_computer_move(game_state)
+                best_move = best_move_and_score[0]
+                print('Suggested move row : ', best_move[0] + 1, 'column : ', best_move[1] + 1)
+            
+            move = get_human_move(game_state)
+            
+            game_state = game_state.make_move(move)
+            number_of_moves = number_of_moves + 1 
+            print('board = ', game_state.get_board(), 'next player = ', game_state.get_player_next())
+
+        else:
+        
+            # player_next_first = game_state.get_player_next()
+            # look_forward = choose_look_ahead()
+            # sort_depth = get_sort_depth()
+            score_count = 0
+            # best_move_and_score = minimax(game_state, look_forward, player_next_first, -inf, inf, sort_depth)
+            best_move_and_score = get_computer_move(game_state)
+            print ('best move and score: ', best_move_and_score, 'Score count : ', score_count)
+            # print('board = ', game_state.get_board(), 'next player = ', game_state.get_player_next())
+            best_move = best_move_and_score[0]
+            game_state = game_state.make_move(best_move)
+            number_of_moves = number_of_moves + 1 
+            print('board = ', game_state.get_board(), 'next player = ', game_state.get_player_next())
+        
+        while undo_move():
+            game_state = game_state.unmake_move()
+            number_of_moves = number_of_moves - 1
+            print('board = ', game_state.get_board(), 'next player = ', game_state.get_player_next())
+            if number_of_moves == 0:
+                break
+            
+        # print ('Game end : ', game_state.game_end())
+        if game_state.game_end():
+            print('End of Game')
+            score_count = 0
+            end_score = game_state.score()
+            if game_state.get_player_next() == 1:

```

Comment:
Might be a good exercise to try and collapse these into one set of if-statements rather than two that look very similar. A main trick here is that when treated as integers, True is equal to 1 and False is equal to 0. So for example I could do things like `print(f'Winner is Player {1 + (winning_player == 2)}')`

Reply:
I've also introduced f-strings above which I see you're not using. Not necessary right now but really nice to use when you want to get more specific about what you're printing